### PR TITLE
feat: note-editor with color palette

### PR DESCRIPTION
**Pull request by PuneetP16**

### CODE REVIEW

File: src/components/UI/Aside/Aside.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+.nav__list_item:hover {
+	color: var(--COLOR-PRIMARY);
+}
+
+/* .notes_nav i {
+} */
+
+/* .notes_nav .nav__item { */
+	/* text */
+/* } */
```
Comment (saurabhchirde) [inline src/components/UI/Aside/Aside.css:36]: Can remove unnecessary comments.

File: src/components/UI/Input/ResizableTextArea.jsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	return (
+		<div>
+			<textarea
+				// rows={textareaheight}
+				onChange={onChange}
+				placeholder={placeholder}
+				className={className}
+				name={name}
+				value={value}
+			></textarea>
```
Comment (saurabhchirde) [inline src/components/UI/Input/ResizableTextArea.jsx:35]: Can add some paddings to your textarea, as it's more congested to use.

File: src/components/UI/Modal/Modal.jsx [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+							onClick={handleClick}
+							key={c.color}
+							className={c.className}
+						></li>
+					);
+				})}
+			</ul>
+		</div>,
+		document.getElementById("react__portal")
+	);
```
Comment (saurabhchirde) [inline src/components/UI/Modal/Modal.jsx:23]: Liked the use of createportal for Modals. Because many people are not aware of this.

**Review by saurabhchirde — APPROVED** [review]
Loved the smooth transition while changing the theme to dark and vice versa and the color palette you have created for notes, it is so minimal and easy to use. 
Other than this, gave some suggestions on some improvement, else Good to merge.

File: src/components/UI/Aside/Aside.css (hunk 1 above)
Comment (PuneetP16) [reply src/components/UI/Aside/Aside.css:36]: will do that

File: src/components/UI/Input/ResizableTextArea.jsx (hunk 2 above)
Comment (PuneetP16) [reply src/components/UI/Input/ResizableTextArea.jsx:35]: Will style it.

File: src/components/UI/Modal/Modal.jsx (hunk 3 above)
Comment (PuneetP16) [reply src/components/UI/Modal/Modal.jsx:23]: Yeah it abstract the modal for ease of implementation.

File: src/reducers/formReducer.js [hunk 4]
```diff
@@ -1,18 +1,29 @@
 export const formReducer = ({ loginData, userData }, action) => {
 	switch (action.type) {
 		case "HANDLE_LOGIN_INPUT":
+			console.log("From input", action);
```
Comment (m26unkwn) [inline src/reducers/formReducer.js:4]: You can destruct the action as you are doing most of the time.

**Review by m26unkwn — APPROVED** [review]
The approachs that I like.

1. Making objects for the icons.
2. Alert component and its state
3.  Wonderful use of objects

otherwise, everything is good and you can just look to the suggestions

**Pull request merged**